simdjson/__init__.pyi

```diff
@@ -0,0 +1,108 @@
+import json
+from typing import (
+    AbstractSet,
+    Any,
+    Dict,
+    Final,
+    Iterator,
+    List,
+    Literal,
+    MutableMapping,
+    NewType,
+    Optional,
+    Tuple,
+    Union,
+    ValuesView,
+)
+
+
+Primitives = Union[int, float, str, boolean]
+K = str
+V = Union['Object', 'Array', Primitives]
+
+
+class Object(Mapping[K, V]):
+  
+    def __getitem__(self, key: K) -> V:
+        ...
+
+    def __iter__(self) -> Iterator[V]:
+        ...
+
+    def __len__(self) -> int:
+        ...
+
+    def as_dict(self) -> Dict[K, V]:
+        ...
+
+    def at_pointer(self, key: str) -> V:
+        ...
+
+    def get(self, key: K, default: Optional[V] = None) -> V:
+        ...
+
+    def keys(self) -> AbstractSet[K]:
+        ...
+
+    def values(self) -> ValuesView[V]:
+        ...
+
+    def items(self) -> AbstractSet[Tuple[K, V]]:
+        ...
+
+    @property
+    def mini(self) -> 'Object':
+        ...
+
+
+Buffer = NewType('Buffer', bytes)
+
+
+class Array(Sequence[V]):
+    def count(self, v: V) -> int:
+        ...
+
+    def as_list(self) -> List[V]:
```

Comment:
`as_list` and `as_object` should probably be `List[Primitives]`, they don't return proxies unlike everything else.

Reply:
Hmm...shouldn't it actually be `as_list -> List[Optional[Union[Primitives, dict, list]]]` and `as_dict -> Dict[str, Optional[Union[Primitives, dict, list]]]` ?

Reply:
PS. I've written in slightly more readable fashion in code.